google/cloud/spanner/internal/clock.h

```diff
@@ -0,0 +1,78 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INTERNAL_CLOCK_H
+#define GOOGLE_CLOUD_CPP_SPANNER_GOOGLE_CLOUD_SPANNER_INTERNAL_CLOCK_H
+
+#include "google/cloud/spanner/version.h"
+#include <chrono>
+#include <mutex>
+
+namespace google {
+namespace cloud {
+namespace spanner {
+inline namespace SPANNER_CLIENT_NS {
+namespace internal {
+
+/**
+ * A simple `Clock` class that can be overridden for testing.
+ *
+ * All implementations of this class are required to be thread-safe.
+ */
+class Clock {
+ public:
+  using time_point = std::chrono::steady_clock::time_point;
+  using duration = std::chrono::steady_clock::duration;
+
+  virtual ~Clock() = default;
+  virtual time_point Now() = 0;
+};
+
+class RealClock : public Clock {
+ public:
+  time_point Now() override { return std::chrono::steady_clock::now(); }
+};
+
+class FakeClock : public Clock {
+ public:
+  /// Construct a `FakeClock` with an optional starting `time_point` of `now`.
+  explicit FakeClock(time_point now = {}) { now_ = now; }
```

Comment:
Given the choice of `steady_clock::time_point`, and the insignificance of its absolute values, I question the need for the ctor argument and `SetTime()`.